components/producers/test_utils.go

```diff
@@ -0,0 +1,70 @@
+package producers
```

Comment:
this is a good idea, please open an RFC PR or a ticket or "something" where you document a testing approach.
we can then all pitch in and ensure that current and future components have the correct structure they need to in order to be testable